tests/test_validators.py

```diff
@@ -1348,15 +1348,23 @@ class Model(BaseModel):
     ]
 
 
-@pytest.mark.xfail()
-def test_partialmethod_as_validator():
-    def custom_validator(additional_stuff: str, cls, values):
-        print(additional_stuff)
-        return values
+@pytest.mark.parametrize('func,allow_reuse', product([partialmethod, partial], [True, False]))
+def test_functool_as_validator(
+    func: Callable,
+    allow_reuse: bool,
+):
```

Comment:
I'm a bit confused by all the magic here, could we simplify this test?

Reply:
I could write the 4 cases as a list I guess

It's just testing both `functools.partial` and `functools.partialmethod` with `allow_reuse` on and off

Reply:
Are the testcases easier to understand with the new ids?